flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -596,32 +651,18 @@ public ReadableConfig getOptions() {
      *
      * @see #createTableFactoryHelper(DynamicTableFactory, DynamicTableFactory.Context)
      */
-    public static class TableFactoryHelper {
-
-        private final DynamicTableFactory tableFactory;
+    public static class TableFactoryHelper extends FactoryHelper<DynamicTableFactory> {
```

Comment:
Unlike `CatalogFactoryHelper`, this wasn't `@PublicEvolving` before, but it seems like it should be as well.